### DISCUSSION TITLE
Notes Jun 8th, 2022

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre#34] [Technical Steering Committee Meetings]
## Organizational Questions
[@wipfli] What should we do with stale repos / dead forks like maplibre-java? https://github.com/maplibre/maplibre-java/issues/1. -> Should add a note in the readme that the repo is currently unmaintained and that we are looking for a maintainer.

[@wipfli] Definition of Voting Members https://github.com/maplibre/maplibre/discussions/32. -> Everyone thinks that the concept is right. @nyurik  suggested that the Governing Board does not need to approve new Voting Members. GitHub Polls https://gh-polls.com/ might be a good way to do the elections and votes.

[@HarelM] Gave a presentation about MapLibre at an Open Source GIS convention. -> Thanks, Harel!

![image](https://user-images.githubusercontent.com/53421382/172773705-1b543704-ce7a-4089-b15a-e08b483902a7.png)

[@nyurik] [Martin project](https://github.com/maplibre/maplibre/issues/31) has migrated from Urbica to MapLibre -> Welcome Martin, we are excited that you joined :)

## MapLibre GL JS Technical Questions
[@HarelM ] Terrain3D has the following main issues which we should address before we can publish a regular release:
- [Memory 1291](https://github.com/maplibre/maplibre-gl-js/issues/1291)
- [Closest tiles 1241](https://github.com/maplibre/maplibre-gl-js/issues/1241)

@prozessor13 has an idea how to fix [#1291](https://github.com/maplibre/maplibre-gl-js/issues/1291) and will follow-up on this.

[@nyurik] [User expressions 1295](https://github.com/maplibre/maplibre-gl-js/issues/1295) -> Discussed allowing referencing run-time callbacks in the style which could do data transformations on feature attributes much like expressions do, but with full javascript support. Americana's [creative usage of the missing style image](https://github.com/ZeLonewolf/openstreetmap-americana/blob/eb9d7ffbcf35c6dcee1550820a182afeac297197/style/americana.js#L477-L479) was mentioned by @mojodna as a way to get run-time callbacks from the style. 

## MapLibre GL Native Technical Questions

[@wipfli] Should we split up maplibre-gl-native?
 - https://github.com/maplibre/maplibre-gl-native/tree/main/platform
 - Should we drop the "gl", because the Apple SDKs might move to Vulkan or Metal?
 - Call the core maplibre-native-core?
 - Have maplibre-native-qt? Have maplibre-native-node? Have maplibre-native-linux? 

-> @nyurik advocated that a monorepo is simpler to maintain than multiple split repos because CI will tell directly side-effects of your changes to the other platforms. @atierian found that the same can be achieved if CI in one repo triggers CI in another repo. @HarelM said that having to take care of side-effects in a different platform which is maybe broken for unrelated reasons discourages people from contributing.

## Next Meeting

I think we forgot to define a date for the next meeting. Should we do it Tuesday, Jul 5th, 2022 at 8:05 PM Zurich-time?

## Attendees (@GitHub handles)
@klokan
@lseelenbinder
@mojodna
@thehoneymad
@atierian
@wipfli
@HarelM
@nyurik

**Comment by wipfli** [comment]
Ah, and as always, your feedback is welcome!

**Comment by HarelM** [comment]
Should be July 13th - the second Wednesday of the month.
Thanks @wipfli for posting these discussions!

**Reply by wipfli** [reply]
July 12th is a Tuesday. Then I think it should be Wednesday, July 13th, 2022.

**Reply by HarelM** [reply]
Yup. :-)

**Reply by wipfli** [reply]
I posted the link to tomorrow's meeting in the Slack channel.

Join the #maplibre slack channel at OSMUS: get an invite at https://osmus-slack.herokuapp.com/
